### PR TITLE
feat: add new workspace scoped key roles

### PR DESCRIPTION
**Pull request by chronark**
also finally make role evaluation typesafe, lol

### CODE REVIEW

File: apps/api/src/routes/legacy_apis_deleteApi.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { rootKeyAuth } from "@/pkg/auth/root_key";
 import { UnkeyApiError, openApiErrorResponses } from "@/pkg/errors";
 import { schema } from "@unkey/db";
 import { newId } from "@unkey/id";
-import { buildQuery } from "@unkey/rbac";
```
Comment (chronark) [inline apps/api/src/routes/legacy_apis_deleteApi.ts:9]: this is typesafe now

Comment (chronark) [reply apps/api/src/routes/legacy_apis_deleteApi.ts:9]: specifically typesafe for our own permissions

it's still generic and we will use the `buildQuery` for our sdk

File: apps/api/src/routes/v1_keys_deleteKey.ts [hunk 2]
```diff
@@ -86,7 +86,7 @@ export const registerV1KeysDeleteKey = (app: App) =>
 
     const auth = await rootKeyAuth(
       c,
-      buildQuery(({ or }) => or("*", `api.${data.api.id}.delete_key`)),
```
Comment (chronark) [inline apps/api/src/routes/v1_keys_deleteKey.ts:89]: forgot this last time

File: apps/web/app/(authenticated)/(app)/app/settings/root-keys/[keyId]/permissions/permissions.ts [hunk 3]
```diff
@@ -8,23 +8,43 @@ type UnkeyPermissions = {
 };
 
 export const workspacePermissions = {
```
Comment (chronark) [inline apps/web/app/(authenticated)/(app)/app/settings/root-keys/[keyId]/permissions/permissions.ts:10]: split this into 2 sections just like the per-api card and also added the missing permissions

CleanShot 2024-01-29 at 10 36 41@2x

File: apps/web/app/(authenticated)/(app)/app/settings/root-keys/[keyId]/permissions/workspace.tsx [hunk 4]
```diff
@@ -18,19 +18,26 @@ export const Workspace: React.FC<Props> = ({ keyId, permissions }) => {
         <CardDescription>Manage workspace permissions</CardDescription>
       </CardHeader>
       <CardContent>
-        <div className="flex flex-col gap-1">
```
Comment (chronark) [inline apps/web/app/(authenticated)/(app)/app/settings/root-keys/[keyId]/permissions/workspace.tsx:21]: updated ui to reflect the previous change

File: internal/rbac/src/queries.ts [hunk 5]
```diff
@@ -29,10 +29,8 @@ export const permissionQuerySchema: z.ZodType<NestedQuery> = z.union([
   }),
 ]);
 
-type Operation = (...args: NestedQuery[]) => NestedQuery;
```
Comment (chronark) [inline internal/rbac/src/queries.ts:32]: changed some types around, no logic changes here

**Review by perkinsjr — APPROVED** [review]

**Pull request merged**